ocs_ci/utility/kms.py

```diff
@@ -470,6 +539,40 @@ def remove_vault_policy(self):
             )
         logger.info(f"Vault policy {self.vault_policy_name} deleted")
 
+    def remove_vault_namespace(self):
+        """
+        Cleanup the namespace
+
+        Raises:
+            KMSResourceCleanupError exception
+
+        """
+        # Unset namespace from environment
+        # else delete will look for namespace within namespace
+        os.environ.pop("VAULT_NAMESPACE")
+        cmd = f"vault namespace delete {self.vault_namespace}/"
+        subprocess.check_output(shlex.split(cmd))
+        if self.vault_namespace_exists(self.vault_namespace):
+            raise KMSResourceCleaneupError(
+                f"Namespace {self.vault_namespace} deletion failed"
+            )
+        logger.info(f"Vault namespace {self.vault_namespace} deleted")
+
+    def get_vault_namespace(self):
+        """
+        From kms connection details resource obtain
+        namespace
+
+        Raises:
+            KMSConnectionDetailsError exception
```

Comment:
```suggestion
            KMSConnectionDetailsError: If unable to retrieve namespace from KMS connection details.
```

Reply:
moved exception out of this function to avoid teardown failures  and handled it in the cleanup() function.